pinot-clients/pinot-jdbc-client/src/main/java/org/apache/pinot/client/PinotDriver.java

```diff
@@ -68,6 +69,9 @@ public Connection connect(String url, Properties info)
       PinotClientTransport pinotClientTransport = factory.buildTransport();
       String controllerUrl = DriverUtils.getControllerFromURL(url);
       String tenant = info.getProperty(INFO_TENANT, DEFAULT_TENANT);
+
+      if(!headers.isEmpty())
+        new PinotConnection(controllerUrl, pinotClientTransport, tenant, new PinotControllerTransport(headers));
```

Comment:
Since the headers are already set into the client factory (line 66), why do we need to add it again?

Reply:
@Jackie-Jiang you can see the issue on this Slack thread - https://apache-pinot.slack.com/archives/CDRCA57FC/p1636968098246300